classes/migration/upgrade/v3_4_0/UpgradeMigration.inc.php

```diff
@@ -80,6 +86,67 @@ public function up(): void
                 $table->dropColumn('setting_type');
             });
         }
+
+        // pkp/pkp-lib#6073: Delete SubmissionFiles/Notes entries that correspond to nonexistent queries.
+        $orphanedIds = DB::table('notes AS n')
+            ->leftJoin('queries AS q', 'n.assoc_id', '=', 'q.query_id')
+            ->where('n.assoc_type', '=', self::ASSOC_TYPE_QUERY)
+            ->whereNull('q.query_id')
+            ->pluck('n.note_id', 'n.assoc_id');
+
+        foreach ($orphanedIds as $neQueryId => $noteId) {
+            error_log("Removing submission files that relates to the entry ID ${noteId} which will be deleted as orphan object");
+            $notesFileRows = DB::table('submission_files as sf')
+                ->join('files as f', 'sf.file_id', '=', 'f.file_id')
+                ->where('sf.assoc_id', '=', $noteId)
```

Comment:
Wherever you query an assoc ID, make sure to also query an assoc type. It'll be ASSOC_TYPE_NOTE in this case, I think.

Reply:
That's correct. Fixed.